src/test/java/com/amazon/opendistroforelasticsearch/security/cache/CachingTest.java

```diff
@@ -103,7 +103,7 @@ public void testRestCachingWithImpersonation() throws Exception {
 
         Assert.assertEquals(4, DummyHTTPAuthenticator.getCount());
         Assert.assertEquals(3, DummyAuthorizer.getCount());
-        Assert.assertEquals(4, DummyAuthenticationBackend.getAuthCount());
+        Assert.assertEquals(1, DummyAuthenticationBackend.getAuthCount());
```

Comment:
Why is this change necessary? It doesn't look related to your change.

Reply:
Same here, caching auth credentials was broken before because equality wasn't working on the key. Now that it is fixed the calls to the authentication backend are properly cached and the backend is hit less.